src/System.Net.HttpListener/src/System/Net/Managed/HttpEndPointManager.cs

```diff
@@ -99,12 +99,18 @@ private static void AddPrefixInternal(string p, HttpListener listener)
                 throw new HttpListenerException((int)HttpStatusCode.BadRequest, SR.net_invalid_path);
 
             // listens on all the interfaces if host name cannot be parsed by IPAddress.
-            HttpEndPointListener epl = GetEPListener(lp.Host, lp.Port, listener, lp.Secure);
+            HttpEndPointListener epl = GetEPListener(lp.Host, lp.Port, listener, lp.Secure, out bool alreadyExists);
+            if (alreadyExists)
+            {
+                throw new HttpListenerException(98, SR.Format(SR.net_listener_already, p));
+            }
             epl.AddPrefix(lp, listener);
         }
 
-        private static HttpEndPointListener GetEPListener(string host, int port, HttpListener listener, bool secure)
+        private static HttpEndPointListener GetEPListener(string host, int port, HttpListener listener, bool secure, out bool alreadyExists)
         {
+            alreadyExists = false;
```

Comment:
I didn't do that in the else block, as `GetEPListener` is called by other methods, such as `RemovePrefixInternal`. This means that we'd throw an exception when trying to remove a prefix that exists, when that's the very thing we actually want to do!

```cs
private static void RemovePrefixInternal(string prefix, HttpListener listener)
{
    ListenerPrefix lp = new ListenerPrefix(prefix);
    if (lp.Path.IndexOf('%') != -1)
        return;

    if (lp.Path.IndexOf("//", StringComparison.Ordinal) != -1)
        return;

    HttpEndPointListener epl = GetEPListener(lp.Host, lp.Port, listener, lp.Secure, out bool dummy);
    epl.RemovePrefix(lp, listener);
}
```